src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs

```diff
@@ -86,10 +86,17 @@ public WebSocketsTransport(WebSocketOptions options, IChannelConnection<byte[]>
 
                 // Wait for the application to finish sending.
                 _logger.WaitingForSend(_connectionId);
-                await sending;
 
-                // Send the server's close frame
-                await socket.CloseOutputAsync(WebSocketCloseStatus.NormalClosure, "", CancellationToken.None);
+                try
+                {
+                    await sending;
```

Comment:
Instead of throwing and catching  a first chance exception, let's write a custom awaiter for this that does .Ignore(). I spoke to @stephentoub and there's a proposal in corefx https://github.com/dotnet/corefx/issues/20732. In the meantime, lets copy that code:

```C#
internal struct NoThrowAwaiter : ICriticalNotifyCompletion
{
    private readonly Task _task;
    public NoThrowAwaiter(Task task) { _task = task; }
    public NoThrowAwaiter GetAwaiter() => this;
    public bool IsCompleted => _task.IsCompleted;
    public void GetResult() { }
    public void OnCompleted(Action continuation) => _task.GetAwaiter().OnCompleted(continuation);
    public void UnsafeOnCompleted(Action continuation) => OnCompleted(continuation);
}
```

And add an extension method .NoThrow to task.